### ISSUE TITLE
Templating variables miss dependencies

### ISSUE THREAD

**Issue by bobrik** [grafana/grafana#13291]
### What Grafana version are you using?

5.2.3

### What datasource are you using?

Prometheus

### What did you do?

I have the following templating variables (there are 7 more, but it's irrelevant):

![image](https://user-images.githubusercontent.com/89186/45569869-b842a700-b815-11e8-8b87-cde417876534.png)

The `instance` is configiured like this:

![image](https://user-images.githubusercontent.com/89186/45569891-c98bb380-b815-11e8-877f-851f5f737d79.png)

I want to change instances when I switch prometheus, the latter is scoped per datacenter.

### What was the expected result?

I switch `prometheus` variable, `instance` refreshes and shows instances from selected `prometheus`.

### What happened instead?

I switch `prometheus` variable, `instance` does not change. When I click on it, it does show the list of instances from the correct `prometheus`.

This in turn confuses other variables down the chain.

**Comment by marefr** [comment]
Works as expected [here](https://play.grafana.org/d/_9-r6LTiz/prometheus-datasource-variable-13291?orgId=1). Don't you agree @bobrik ?

**Comment by hagen1778** [comment]
Hi @bobrik 
I was unable to reproduce the issue. All variables were updated consequentially. It could be an error while parsing variable value before sending request to CH... Can I ask for some additional information:
* are there any errors in console (developer tools>console)?
* can you check outgoing requests to ClickHouse when switching variables (developer tools>network)

We could move to Clickhouse datasource repo with this issue.

**Comment by hagen1778** [comment]
Hi @bobrik @marefr 

I've got the same error, which occurs not every time, while changing datasources at playground dashboard. The gist for dashboard can be found [here](https://gist.github.com/hagen1778/0bd00b0555cf8f4090a9b6195148f6b8)

The screenshot of error:
<img width="1001" alt="screen shot 2018-10-05 at 11 12 35" src="https://user-images.githubusercontent.com/2902918/46524299-0e44c080-c891-11e8-8915-624a5fd86463.png">

We see that last query lacks for variable values `(go_goroutines + )>`. It is because the second query from the screenshot returned empty list: `{"status":"success","data":{"resultType":"vector","result":[]}}`

@bobrik I've noticed that behavior is more stable and predictable on the last Grafana version (Grafana v5.4.0-pre1 (commit: 61d64c4)).

**Comment by marefr** [comment]
@hagen1778 thanks. Imported your gist [here](https://play.grafana.org/d/_9-r6LTiz/prometheus-datasource-variable-13291?orgId=1) but still can't seem to reproduce. Note play.grafana.org are running Grafana v5.3.0-beta3 now. 

Can you describe the steps in detail that leads to your problem? Do you see any errors logged in the console in chrome dev tools? What browser and version are you using?

**Comment by hagen1778** [comment]
Hi @marefr 

My browser Google Chrome Version 69.0.3497.100

The bug appears randomly. I'm still able to reproduce it at playground by changing datasources in `$ds` variable.
What I see in network tab:
1. First query: `match[]: up` // Returns a lot of data
2. Second query: `go_goroutines{instance="172.31.1.182:443"}` // Returns nothing `{"status":"success","data":{"resultType":"vector","result":[]}}`
3. Third query: `(go_goroutines + )>`

Errors in the console:
* angular.js:12759 GET https://play.grafana.org/api/datasources/proxy/29/api/v1/query?query=(go_goroutines%20%2B%20)%3E&time=1538989103 400
* angular.js:14700 Possibly unhandled rejection: {"data":{"status":"error","errorType":"bad_data","error":"parse error at char 18: no valid expression found","message":"parse error at char 18: no valid expression found"},"status":400,"config":{"method":"GET","transformRequest":[null],"transformResponse":[null],"jsonpCallbackParam":"callback","url":"api/datasources/proxy/29/api/v1/query?query=(go_goroutines%20%2B%20)%3E&time=1538989103","retry":0,"headers":{"X-Grafana-Org-Id":1,"Accept":"application/json, text/plain, */*"}},"statusText":"","xhrStatus":"complete"}

Please note, cache option is disabled in chrome dev tools.

**Comment by marefr** [comment]
Thanks. Able to reproduce now. You can go to this [link](https://play.grafana.org/d/_9-r6LTiz/prometheus-datasource-variable-13291?orgId=1&var-ds=Prometheus&var-instance=10.244.0.7:9091&var-goroutines=&var-filtered_goroutines=336) to reproduce.

Question is what you expect to happen when
> Second query: go_goroutines{instance="172.31.1.182:443"} // Returns nothing {"status":"success","data":{"resultType":"vector","result":[]}}

**Comment by hagen1778** [comment]
Yes, I see. It returns nothing, so variable is empty too. I'm just trying to understand what causes an issue in @bobrik case since I can't reproduce it locally. And I've tried to show that it is possible to have an empty variable value in query requests. Thank you for your time!

**Comment by marefr** [comment]
Thank you. So probably this is something that was enhanced/fixed when #11890 was fixed. It will be included in v5.3.0 stable. Will consider reopen this if @bobrik still experience problem on v5.3.0-beta3.

**Comment by bobrik** [comment]
We've upgraded and it seems to be working, thanks all!
